### ISSUE TITLE
CentOS 7 installation does not work and verbose mode provides no information

### ISSUE THREAD

**Issue by ruhulsbu** [NERSC/shifter#178]
Hi, 
I am trying to install Shifter on CentOS 7. I have followed the installation commands mentioned here <https://github.com/NERSC/shifter/blob/master/doc/install/centos7.rst>. But the installation is not working. I only get the help menu with shifter or shifterimg command. I also do not get any information while trying to pull images from docker. Can you please explain why I am facing such a problem? Also, I see the installation manuals were not updated after the latest release. So, please let me know if I am following the correct installation manual.

-Ruhul Amin
Stony Brook University

**Comment by ruhulsbu** [comment]
Hi SCanon,
Thank you very much for your reply. I have attached the log file of "rpmbuild -tb shifter-$VERSION.tar.gz" command. I would like to know whether I could build the rpms correctly. I also want to know how to test the rpm installation as root. 
After that I would like to test my redis, mongodb, munge and gunicorn installation.

**Attachment:** **[rpmbuild_shifter.txt]**(https://github.com/NERSC/shifter/files/1125859/rpmbuild_shifter.txt)

Thanking You,
-Ruhul
SEAWulf HPC, 
Stony Brook University

**Comment by scanon** [comment]
Ruhul,

Okay, I see the issue.  This has been fixed in the master branch but the 16.08.1 tag still has the bug.  The URL location of one of the dependencies changed.  You can fix this in place with this...

```
diff --git a/dep/build_ssh.sh b/dep/build_ssh.sh
index 5294b96..5c1abb0 100755
--- a/dep/build_ssh.sh
+++ b/dep/build_ssh.sh
@@ -39,7 +39,7 @@ if [[ ! -e "zlib-${ZLIB_VERSION}.tar.gz" && -n "$DEPTAR_DIR" && -e "$DEPTAR_DIR/
     cp "$DEPTAR_DIR/zlib-${ZLIB_VERSION}.tar.gz" .
 fi
 if [[ ! -e "zlib-${ZLIB_VERSION}.tar.gz" ]]; then
-    curl -o "zlib-${ZLIB_VERSION}.tar.gz" "http://zlib.net/zlib-${ZLIB_VERSION}.tar.gz"
+    curl -o "zlib-${ZLIB_VERSION}.tar.gz" "http://zlib.net/fossils/zlib-${ZLIB_VERSION}.tar.gz"
 fi
 if [[ ! -e "openssh-${OPENSSH_VERSION}.tar.gz" && -n "$DEPTAR_DIR" && -e "$DEPTAR_DIR/openssh-${OPENSSH_VERSION}.tar.gz" ]]; then
     cp "$DEPTAR_DIR/openssh-${OPENSSH_VERSION}.tar.gz" .
```

then the build should work.  Alternatively, you can find a prebuilt RPMs here...

http://portal.nersc.gov/project/bigdata/shifter/

You can also look at this Dockerfile for an example of how to install it.

https://raw.githubusercontent.com/NERSC/Shifter-Tutorial/master/examples/shifter-in-a-box/Dockerfile

Hope this helps.  Also you can subscribe and ask these questions on the mailing list...

https://groups.google.com/forum/#!forum/shifter-hpc

**Comment by scanon** [comment]
Also, tag 16.08.3_2 should have these fixes.   We will update the docs to recommend this.
